crates/apollo-compiler/src/validation/mod.rs

```diff
@@ -38,20 +38,32 @@ pub(crate) use validation_db::{ValidationDatabase, ValidationStorage};
 /// Wraps a [`Schema`] or [`ExecutableDocument`] to mark it
 /// as [valid](https://spec.graphql.org/October2021/#sec-Validation).
 ///
-/// This is obtained from one of:
+/// This is obtained either by running validation with one of:
 ///
 /// * [`Schema::parse_and_validate`]
 /// * [`Schema::validate`]
 /// * [`ExecutableDocument::parse_and_validate`]
 /// * [`ExecutableDocument::validate`]
 ///
+/// … or by explicitly skipping it with [`Valid::assert_valid`].
+///
 /// The schema or document inside `Valid<T>` is immutable (`&mut T` is not given out).
 /// It can be extracted with [`into_inner`][Self::into_inner],
 /// such as to mutate it then possibly re-validate it.
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct Valid<T>(pub(crate) T);
 
 impl<T> Valid<T> {
+    /// Construct a `Valid` document without actually running validation.
+    ///
+    /// The caller takes responsibility to ascertain that
+    /// the document is known through some other means to be valid.
+    /// For example, if it was loaded from some external storag
+    /// where it was only stored after validation.
+    pub fn assert_valid(document: T) -> Self {
```

Comment:
With `assert` i kind of expect it to work like `assert!`, what do you think about a different name like `assume_valid()`?

Reply:
I probably came up with this based on https://doc.rust-lang.org/core/panic/struct.AssertUnwindSafe.html, but I think the association with `assert!` and "can panic" is still better avoided. I’ll rename to `assume_valid`